hyp3_metadata/insar.py

```diff
@@ -0,0 +1,145 @@
+        return self.create_metadata_file(self.payload, 'insar/corr_tif.xml.j2', reference_file)
+
+    def create_dem_tif_xml(self) -> Path:
+        reference_file = self.dir / f'{self.file_name}_dem.tif'
+        return self.create_metadata_file(self.payload, 'insar/dem.xml.j2', reference_file)
+
+    def create_los_displacement_xml(self) -> Path:
+        reference_file = self.dir / f'{self.file_name}_los_disp.tif'
+        return self.create_metadata_file(self.payload, 'insar/los_disp_tif.xml.j2', reference_file)
+
+    def create_look_vector_xmls(self) -> List[Path]:
+        reference_file_phi = self.dir / f'{self.file_name}_lv_phi.tif'
+        reference_file_theta = self.dir / f'{self.file_name}_lv_theta.tif'
+        output_files = [
+            self.create_metadata_file(self.payload, 'insar/lv_phi_tif.xml.j2', reference_file_phi),
+            self.create_metadata_file(self.payload, 'insar/lv_theta_tif.xml.j2', reference_file_theta),
+        ]
+        return output_files
+
+    def create_browse_xmls(self) -> Path:
+        reference_file_col = self.dir / f'{self.file_name}_color_phase.png'
+        reference_file_unw = self.dir / f'{self.file_name}_unw_phase.png'
+        output_files = [
+            self.create_metadata_file(self.payload, 'insar/color_phase_png.xml.j2', reference_file_col),
+            self.create_metadata_file(self.payload, 'insar/unw_phase_png.xml.j2', reference_file_unw),
+        ]
+        return output_files
+
+    def create_unwrapped_phase_xml(self) -> Path:
+        reference_file = self.dir / f'{self.file_name}_unw_phase.tif'
+        return self.create_metadata_file(self.payload, 'insar/unw_phase_tif.xml.j2', reference_file)
+
+    def create_vertical_displacement_xml(self) -> Path:
+        reference_file = self.dir / f'{self.file_name}_vert_disp.tif'
+        return self.create_metadata_file(self.payload, 'insar/vert_disp_tif.xml.j2', reference_file)
+
+    def create_wrapped_phase_xml(self) -> Path:
+        reference_file = self.dir / f'{self.file_name}_wrapped_phase.tif'
+        return self.create_metadata_file(self.payload, 'insar/wrapped_phase_tif.xml.j2', reference_file)
+
+    @classmethod
+    def create_metadata_file(cls, payload: dict, template: str, reference_file: Path, out_ext: str = 'xml',
+                             strip_ext: bool = False, name_only=False) -> Optional[Path]:
+        if not reference_file.exists():
+            return None
+
+        payload = deepcopy(payload)
+        info = gdal.Info(str(reference_file), format='json')
+        payload['pixel_spacing'] = info['geoTransform'][1]
+        payload['projection'] = get_projection(info['coordinateSystem']['wkt'])
+
+        payload['thumbnail_encoded_string'] = get_thumbnail_encoded_string(reference_file)
+
+        content = render_template(template, payload)
+        out_name = reference_file.name if not strip_ext else reference_file.stem
+        if name_only:
+            out_name = payload['product_dir'].name
```

Comment:
Both could be combined into a strip suffix option...

```python
strip_suffix = '.tif'
strip_suffix = '_vert_disp.tif'
```
but that'd require knowing the suffix which these don't. 

I want to be able to combine them somehow though...